path_oidc.go

```diff
@@ -71,33 +91,58 @@ func pathOIDC(b *jwtAuthBackend) []*framework.Path {
 				logical.UpdateOperation: &framework.PathOperation{
 					Callback: b.authURL,
 					Summary:  "Request an authorization URL to start an OIDC login flow.",
+
+					// state is cached so don't process OIDC logins on perf standbys
+					ForwardPerformanceStandby: true,
 				},
 			},
 		},
 	}
 }
 
-func (b *jwtAuthBackend) pathCallback(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+func (b *jwtAuthBackend) pathCallbackPost(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	config, err := b.config(ctx, req.Storage)
+	if err != nil {
+		return nil, err
+	}
+	if config == nil {
+		return logical.ErrorResponse(errLoginFailed + " Could not load configuration."), nil
+	}
 
-	// Because the state is cached, don't process OIDC logins on perf standbys
-	if b.System().ReplicationState().HasState(consts.ReplicationPerformanceStandby) {
-		return nil, logical.ErrReadOnly
+	if config.OIDCResponseMode != responseModeFormPost {
+		return logical.RespondWithStatusCode(nil, req, http.StatusMethodNotAllowed)
 	}
 
-	state := b.verifyState(d.Get("state").(string))
-	if state == nil {
-		return logical.ErrorResponse(errLoginFailed + " Expired or missing OAuth state."), nil
+	stateID := d.Get("state").(string)
```

Comment:
Should we be calling `verifyState` anywhere in here? Forgive my lack of familiarity with this codebase, I only visit it from time to time!

Reply:
I suppose it's probably intentionally this way because `amendState` looks very similar, but just want to understand it better.